libs/blocks/preflight/panels/general.js

```diff
@@ -25,7 +30,13 @@ async function getStatus(url) {
   const preview = json.preview.lastModified || DEF_NEVER;
   const live = json.live.lastModified || DEF_NEVER;
   const publish = await userCanPublishPage(json, false);
-  const edit = json.edit.url;
+  let edit = json.edit.url;
+  if (
+    !json.edit.url
+    && json.preview.sourceLocation
+    && json.preview.sourceLocation.includes(DA_DOMAIN)) {
+    edit = getDAEditUrl(json.preview.sourceLocation);
+  }
```

Comment:
I think we can simplify this by doing:
(this will also remove the need for a separate `getDAEditUrl` function and the `DA_DOMAIN` constant)
```suggestion
    const { sourceLocation } = json.preview;
    const edit = json.edit.url
    || (sourceLocation?.includes('da.live') && sourceLocation?.replace('markup:https://content.da.live', 'https://da.live/edit#'))
    || '';
```

Reply:
Implemented this suggestion. Have removed the function getDAEditUrl(...)  but retained the DA_DOMAIN variable since it is also used to select the relevant icon for the edit app (SP or DA). Using explicit string values at multiple places can introduce unwanted human errors later on and becomes tedious to change constant values if needed.